client/my-sites/post-share/index.jsx

```diff
@@ -112,162 +149,296 @@ class PostShare extends Component {
+				>
+					{ translate( 'Add account' ) }
+				</Button>
+			</div>
+		);
+	}
+
+	renderPrimarySection() {
+		const { hasFetchedConnections, siteSlug, translate } = this.props;
+
+		if ( ! hasFetchedConnections ) {
+			return null;
+		}
+
+		if ( ! this.hasConnections() ) {
+			return (
+				<NoConnectionsNotice { ...{
+					siteSlug,
+					translate,
+				} } />
+			);
+		}
+
+		return (
+			<div>
+				<div className="post-share__main">
+					<div className="post-share__form">
+						{ this.renderMessage() }
+						{ this.renderSharingButtons() }
+					</div>
+
+					{ this.renderConnectionsSection() }
+				</div>
+
+				{ this.renderUpgradeToGetSchedulingNudge() }
+				{ this.renderActionsSection() }
+			</div>
+		);
 	}
 
 	render() {
 		if ( ! this.props.isPublicizeEnabled ) {
 			return null;
 		}
 
-		if ( ! this.props.planHasRepublicizeFeature ) {
-			return ( <div className="post-share">
-				<Banner
-					feature="republicize"
-					title={ this.props.translate( 'Unlock the ability to re-share posts to social media' ) }
-					callToAction={ this.props.translate( 'Upgrade to Premium' ) }
-					description={ this.props.translate( 'Get unlimited premium themes, video uploads, monetize your site and more.' ) }
-				/>
-			</div> );
+		if ( ! this.props.hasRepublicizeFeature ) {
+			return this.renderUpgradeToGetPublicizeNudge();
+		}
+
+		const {
+			message,
+			hasRepublicizeFeature,
+			hasRepublicizeSchedulingFeature,
+			postId,
+			siteId,
+			siteSlug,
+			translate,
+		} = this.props;
+
+		if ( ! siteId || ! postId ) {
+			return null;
 		}
 
-		const classes = classNames( 'post-share__wrapper', {
-			'has-connections': this.hasConnections()
-		} );
+		const classes = classNames(
+			'post-share__wrapper',
+			{ 'has-connections': this.hasConnections() },
+			{ 'has-republicize-feature': hasRepublicizeFeature },
+			{ 'has-republicize-scheduling-feature': hasRepublicizeSchedulingFeature },
+		);
 
 		return (
 			<div className="post-share">
-				{ this.props.requesting &&
-					<Notice status="is-warning" showDismiss={ false }>
-							{ this.props.translate( 'Sharing…' ) }
-					</Notice>
-				}
-
-				{ this.props.success &&
-					<Notice status="is-success" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Post shared. Please check your social media accounts.' ) }
-					</Notice>
-				}
+				<QueryPostTypes siteId={ siteId } />
+				<QueryPublicizeConnections siteId={ siteId } />
 
-				{ this.props.failure &&
-					<Notice status="is-error" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Something went wrong. Please don\'t be mad.' ) }
-						</Notice>
-				}
+				{ this.renderRequestSharingNotice() }
 
 				<div className={ classes }>
-					{ this.props.siteId && <QueryPostTypes siteId={ this.props.siteId } /> }
 					<div className="post-share__head">
 						<h4 className="post-share__title">
-							{ this.props.translate( 'Share post' ) }
+							{ translate( 'Share this post' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( 'Share this post!' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).